core/src/main/spark-2.4/org/apache/spark/sql/TiStrategy.scala

```diff
@@ -253,6 +250,19 @@ case class TiStrategy(getOrCreateTiContext: SparkSession => TiContext)(sparkSess
       tableScanPlan
     }
 
+    if (tiColumns.isEmpty) {
```

Comment:
Since it is used in both spark 2.3 and spark 2.4, maybe we can wrap this logic into a function?